library/std/src/thread/mod.rs

```diff
@@ -1761,6 +1761,18 @@ impl<T> JoinHandle<T> {
     /// }).unwrap();
     /// join_handle.join().expect("Couldn't join on the associated thread");
     /// ```
+    ///
+    /// # Notes
+    ///
+    /// This function has the same minimal guarantee regarding "foreign" unwinding operations (e.g.
+    /// an exception thrown from C++ code, or a `panic!` in Rust code compiled or linked with a
+    /// different runtime) as [`catch_unwind`]; namely, catching such an exception using this
+    /// function will have one of two behaviors, and it is unspecified which will occur:
```

Comment:
```suggestion
    /// different runtime) as [`catch_unwind`]; namely, if the thread unwinds all the way
    /// to the root with such an exception, this
    /// function will have one of two behaviors, and it is unspecified which will occur:
```

Reply:
Isn't catching "using this function" (i.e. `join`) somewhat more specific than "unwind[ing] all the way to the root (of the thread)", since there's no requirement that `join` be called on a spawned thread, and there's no such guarantee if the thread itself is spawned in C or C++?

Reply:
I found "catch using this function" rather confusing since `join` isn't even doing any catching. The catching is done by the root function of the new thread installed by `thread::spawn`, which then communicates with `join` to forward the caught payload.

How would this ever be called on a thread spawned by another runtime?

The argument that join might not be called is an argument for my change, isn't it? If a foreign exception propagates to the root of a `thread::spawn` thread, that may abort the process even if `join` is never called.

Reply:
Ah, okay, so if `thread::spawn` is what actually causes the `catch_unwind` to be called, I'll change it to mention that rather than referring to "this function".